code/modules/mob/living/simple_animal/hostile/mining_mobs.dm

```diff
@@ -734,8 +734,76 @@
 	icon = 'icons/obj/surgery.dmi'
 	icon_state = "demon_heart"
 
+//Lavaland Surface Boss Gol'goth-Een the Ancient
+
+/mob/living/simple_animal/hostile/asteroid/goliath/beast/boss
+	name = "Gol'goth-Een the Ancient"
+	desc = "If a goliath is massive... this thing is bigger than massive."
+	damage_coeff = list(BRUTE = 0, BURN = 0, TOX = 0, CLONE = 0, STAMINA = 0, OXY = 0) //Our weapons are useless!
+	//Health not listed because it can not be defeated through damage
+	melee_damage_lower = 40
+	melee_damage_upper = 40
+	del_on_death = 1
+	var/obj/structure/lavaland_door/gate
+	var/tendrilSpawnCooldown = 0
+	var/currentSpawnTime = 10
+	var/TENDRILSPAWNTIMEMAX = 15
+	var/TENDRILSPAWNTIMEMIN = 10
+
+/mob/living/simple_animal/hostile/asteroid/goliath/beast/boss/New()
+	..()
+	transform *= 2
+	for(var/Pl in player_list)
+		var/mob/P = Pl
+		if(P.z == z)
+			P.show_message("<span class='danger'><B>You here a massive roar in the distance.</B></span>", 2,"<span class='danger'><B>You feel the force of a shockwave.</B></span>", 1)
+
+/mob/living/simple_animal/hostile/asteroid/goliath/beast/boss/Life()
+	tendrilSpawnCooldown++
+	..()
+
+/mob/living/simple_animal/hostile/asteroid/goliath/beast/boss/OpenFire()
+	if(tendrilSpawnCooldown > currentSpawnTime)
+		visible_message("<span class='danger'>[src] slams the ground causing a tendril to come up beneath [target]!</span>")
```

Comment:
sure but the way you've phrased it sounds like shit when read
is english your first language because if not i dont blame you